nativelib/src/main/resources/posix/sys/socket.c

```diff
@@ -0,0 +1,123 @@
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <string.h>
+#include <stdio.h>
+
+struct scalanative_sockaddr {
+    char sa_data[14];
+    sa_family_t sa_family;
+};
+
+void scalanative_convert_sockaddr(struct scalanative_sockaddr *in,
+                                  struct sockaddr *out) {
+    out->sa_family = in->sa_family;
+    void *ignored = memcpy(out->sa_data, in->sa_data, 14);
```

Comment:
I'm not sure there's a better way to do this for now, but I think this will clip off some data if sockaddr is actually a pointer to a larger struct like sockaddr_in6 or sockaddr_storage.